### ISSUE TITLE
how to pass automatically in offline mode

### ISSUE THREAD

**Issue by davidenitti** [allegroai/clearml#583]
there are cases when the connection to the server is not available and clearml waits (forever?) for the connection, is there a way to pass in offline mode automatically if this happens?
or save offline files even if in online mode, so if the connection is lost the training can continue.
thank you very much for the help!

**Comment by jkhenning** [comment]
Hi @davidenitti ,

While not trivial, this would indeed be a very cool feature 🙂 
I'll mark this as a feature request

**Comment by Talljack** [comment]
Is there any new progress on this feature? Looking forward to using it

**Comment by ainoam** [comment]
Thanks for the friendly nudge @Talljack  - While we haven't yet had a chance to move this forward, it's definitely on our shortlist.

**Comment by Talljack** [comment]
> Thanks for the friendly nudge @Talljack - While we haven't yet had a chance to move this forward, it's definitely on our shortlist.

Thanks for your reply. I want to ask, I added a ping interface to the Session, and then called Task.set_offline in the StopSignal test, but my task cannot end normally, even if the close method is called, because my other add_batch interfaces are always retrying. Is there any way to solve it?
![image](https://github.com/user-attachments/assets/f38f81fb-6a4a-4162-b403-a6b296cc4623)

**Comment by jkhenning** [comment]
One option is to wait until all events are flushed, I think

**Comment by Talljack** [comment]
> One option is to wait until all events are flushed, I think

Good idea, when my task training is finished and the model is saved locally, but the backend interface keeps retrying and cannot be completed, how can I know that all events are flushed? Or what should I do? Thank you very much for your reply!
